starport/services/chain/build.go

```diff
@@ -52,7 +55,13 @@ func (c *Chain) build(ctx context.Context) (err error) {
 		return err
 	}
 
-	return gocmd.InstallAll(ctx, c.app.Path, buildFlags)
+	binary, err := c.Binary()
+	if err != nil {
+		return err
+	}
+
+	cmdPath := filepath.Join(c.app.Path, cmdFolder, c.app.D())
```

Comment:
One feature of `go install` I was liking that it is able to auto discover the location of the main package. This way `starport build` command always work even if the main package's location is not under `cmd/`.

can we accommodate that by making main package discovery with code analysis? by creating a `pkg/goanalysis.DiscoverMain(rootPath string) (paths []string, err error)`.

Reply:
done on [`a59c78d`](https://github.com/tendermint/starport/pull/1521/commits/a59c78db4f9d6ffa1469b54049b8dce55aa72c17)